tests/unit/DateUtilsTest.js

```diff
@@ -35,17 +47,70 @@ describe('DateUtils', () => {
 
         const date = moment.utc('2022-11-05').set({hour: 10, minute: 17});
         expect(DateUtils.datetimeToCalendarTime(LOCALE, date)).toBe('Nov 5, 2022 at 10:17 AM');
+
+        const todayLowercaseDate = moment.utc().set({hour: 14, minute: 32});
+        expect(DateUtils.datetimeToCalendarTime(LOCALE, todayLowercaseDate, false, undefined, true)).toBe('today at 2:32 PM');
+    });
+
+    it('should update timezone if automatic and selected timezone do not match', () => {
+        Intl.DateTimeFormat = jest.fn(() => ({
+            resolvedOptions: () => ({timeZone: 'America/Chicago'}),
+        }));
+        Onyx.set(ONYXKEYS.PERSONAL_DETAILS_LIST, {999: {timezone: {selected: 'Etc/UTC', automatic: true}}}).then(() => {
+            const result = DateUtils.getCurrentTimezone();
+            expect(result).toEqual({
+                selected: 'America/Chicago',
+                automatic: true,
+            });
+        });
+    });
+
+    it('should not update timezone if automatic and selected timezone match', () => {
+        Intl.DateTimeFormat = jest.fn(() => ({
+            resolvedOptions: () => ({timeZone: 'Etc/UTC'}),
+        }));
+        Onyx.set(ONYXKEYS.PERSONAL_DETAILS_LIST, {999: {timezone: {selected: 'Etc/UTC', automatic: true}}}).then(() => {
```

Comment:
```suggestion
        Onyx.set(ONYXKEYS.PERSONAL_DETAILS_LIST, {999: {timezone: {selected: 'UTC', automatic: true}}}).then(() => {
```

I think it can be just `UTC` (please test)